presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -605,7 +605,7 @@ private PlanNode projectCoercionsIfNeeded(TranslationMap translations, PlanNode
     private Optional<Expression> tryResolveMissingExpression(PlanBuilder subPlan, Expression expression)
     {
         Expression rewritten = subPlan.rewrite(expression);
-        if (rewritten instanceof SymbolReference) {
+        if (rewritten instanceof SymbolReference || rewritten instanceof Cast) {
```

Comment:
Maybe this would be enough
```
 if (rewritten != expression)
```

If not I think you need to check if child of `Cast` is an instance of `SymbolReference`